### ISSUE TITLE
Avoid mutating arguments to functions

### ISSUE THREAD

**Issue by gaand** [ga-wdi-boston/js-functions-ins-and-outs#32]
Frequently suboptimal.

**Comment by jrhorn424** [comment]
Did you mean we should note that mutating reference-type arguments is an anti-pattern? It's a large part of the talk.

**Comment by gaand** [comment]
Yes, but perhaps we could reduce that.  At least note that a mutator should be explicit.  So, something that only works as a mutator is one thing.  Perhaps a forward reference to the `!` modifier in ruby?  Not all things that mutate have it, but that's because they don't have a sensible non-mutator form.
